### ISSUE TITLE
Compiler Error

### ISSUE THREAD

**Issue by jake-fishtech** [supersimple33/Touch-Bar-Visualizer#1]
Hello!

I tried to build this project in xcode, but it seems there might be a few issues. Here is what pops up when I hit build.

"No such module AMCoreAudio". 
"Expression implicitly coerced from Float to any" (volume.swift)
"Variable 'cPnt' was never mutated"
"Variable 'cPntLen' was never mutated"
"Immutable value 'i' was never used"
"Will never be executed" (volume.swift)

PS. I have been looking and attempting to build something like this for quite some time. I love this project and would LOVE to talk to you about it and feature it on my YouTube channel. If you would be willing to talk to me, please send an email to jake.fishtech@gmail.com thanks =)

<img width="1399" alt="Screen Shot 2020-07-20 at 7 08 40 PM" src="https://user-images.githubusercontent.com/65430128/87995576-e3574b00-cabd-11ea-912f-955e1acc7ef8.png">

**Comment by jake-fishtech** [comment]
I found the binary version in the pre-releases section - and that works perfectly well! Still would love to talk to you about this project and where you see it going in the future. 

Best, Jake
jake.fishtech@gmail.com

**Comment by supersimple33** [comment]
Hey Jake I would love to help you here. In regard to the `No such module AMCoreAudio` are you opening the TouchBarVisualizer.xcworkspace or the TouchBarVisualizer.xcodeproj file. You must build and run with the pods found in the xcworkspace file in order to be able to import AMCoreAudio.

**Comment by supersimple33** [comment]
To be a little more specific you must build and run the Pods-TouchBarVissualizer and AMCoreAudio frameworks in order for Xcode to recognize the frameworks existence. (You should find them in the drop-down menu next to the start and stop program debugging in the top left.

**Comment by jake-fishtech** [comment]
Ah, that fixed it! It now will build successfully, but with a few fatal errors. It will attempt to launch the visualizer, but it will become unresponsive once it tries to launch. The fatal error that shows up in a few places is "Thread 1: Fatal error: Unexpectedly found nil while unwrapping an Optional value". Thank you so much for the help!

**Comment by supersimple33** [comment]
I believe the error you are talking about was on line 72 of ViewController.swift the newest commit I pushed should resolve that error.

**Comment by jake-fishtech** [comment]
That fixed it! Thank you so much for the help! I did notice a few bugs when using the app. If you close the app with the "x" on the touch bar and then try to reopen it, the visualizer stops working and looks like this:
<img width="1085" alt="Touch Bar Shot 2020-07-29 at 3 43 46 PM" src="https://user-images.githubusercontent.com/65430128/88845551-58a7d780-d1b2-11ea-8d24-fee8d9640cdc.png">
I also found another bug where if the audio output is changed away from "TVB" while the app is open, when you reopen the app later, it doesn't change to the TVB output again.

Just some bugs I found, if you weren't aware of them before :). Thank you again for the help.
I was wondering if I could contact you about this app - if your okay with that, is there an email I could reach you at?
